### PR TITLE
Fix Downlevel Vertex Stage Storage Buffer Check

### PR DESCRIPTION
**Pull request by cwfitzgerald**
**Connections**

Fixes a bug in #1599. Also follows up on #1583 marking all of hello-compute as a failure.

**Testing**

Tests now pass on rpi4.

### CODE REVIEW

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -981,17 +981,17 @@ impl<A: HalApi> Device<A> {
...
             let mut required_features = wgt::Features::empty();
             let mut required_downlevel_flags = wgt::DownlevelFlags::empty();
-            let (array_feature, is_writable_storage) = match entry.ty {
+            let (array_feature, storage_buffer, is_writable_storage) = match entry.ty {
                 Bt::Buffer {
                     ty: wgt::BufferBindingType::Uniform,
                     has_dynamic_offset: false,
                     min_binding_size: _,
-                } => (Some(wgt::Features::BUFFER_BINDING_ARRAY), false),
+                } => (Some(wgt::Features::BUFFER_BINDING_ARRAY), false, false),
```
Comment (kvark) [inline wgpu-core/src/device/mod.rs:989]: Would you mind turning this into an enum? booleans in such position are bad, and having two of them is twice as bad:)

**Comment by kvark** [comment]
The SSBO in VS support is pretty sad story - https://www.raspberrypi.org/forums/viewtopic.php?t=271863
Here is my reading of this:
  1. Tiling GPUs don't want *writable* storage resources that can be tested.
  2. Regular writable SSBOs and images can *not* be tested, because accessing them within the same draw call is technically UB.
  3. Only atomic access to SSBO can be tested, and that's what the drivers do *not* want us to do.
  4. GLES doesn't have a proper limit to express this, but Vulkan and WebGPU do. We have `VERTEX_WRITABLE_STORAGE` that captures this much better.
  5. Perhaps we can detect if 0 is returned for `MAX_VERTEX_SHADER_STORAGE_BLOCKS` and ignore it (while not exposing `VERTEX_WRITABLE_STORAGE`)? Unclear what to do if there is a GL error. May need to hack a workaround in the error callback for this specifically :(

**Comment by cwfitzgerald** [comment]
Alright, this is good to go on my machine

**Review by kvark — COMMENTED** [review]
Great, thank you!
bors r+

**Pull request merged**